### PR TITLE
tools: add large binlog operation faq

### PR DESCRIPTION
**Pull request by amyangfei**
### What is changed, added or deleted? 

add DM operation FAQ for large binlog interruption

### Which version does your change affect? 

dev, v3.0

### Checklist 

- [x] No Tab spaces anywhere 
- [x] Keep the first level heading consistent with `title` in metadata
- [x] Use *four* spaces for each level of indentation except that in `TOC.md`

### CODE REVIEW

**Comment by amyangfei** [comment]
PTAL @GregoryIan @csuzhangxc

File: dev/faq/data-migration.md [hunk 1]
```diff
@@ -19,3 +19,10 @@ category: FAQ
 发生 `driver: bad connection` 错误时，通常表示 DM 到下游 TiDB 的数据库连接出现了异常（如网络故障、TiDB 重启等）且当前请求的数据暂时未能发送到 TiDB。
 
 当前版本 DM 发生该类型错误时，需要先使用 `stop-task` 停止任务后再使用 `start-task` 重启任务。后续 DM 会完善对此错误类型的自动重试机制。
+
+## 同步任务中断并包含 `get binlog error ERROR 1236 (HY000)`, `binlog checksum mismatch, data may be corrupted` 等 binlog 获取或解析失败错误
+
+在 DM 进行增量同步过程中，如果遇到了上游超过 4G 的 binlog 文件，并且 DM 在处理这个 binlog 过程中出现了同步中断（包括正常的 pause/stop 任务，异常的原因发生的中断时），DM 需要保存同步 checkpoint 信息，因为 MySQL position 使用 uint32 存储，所以超过 4G 部分的断点位置值会溢出，存储的是一个不合法的断点值，使用该值重新解析 binlog/relay log 就会出错，遇到这种情况需要手动进行恢复。恢复方法：
```
Comment (GregoryIan) [inline dev/faq/data-migration.md]:
```suggestion
在 DM 进行增量同步过程中，如果遇到了上游超过 4G 的 binlog 文件，并且 DM 在处理这个 binlog 过程中出现了同步中断（包括正常的 pause/stop 任务，异常的原因发生的中断时）, 就可能出现这个错误。原因是 DM 需要保存同步的 checkpoint 信息，但是 MySQL position 官方定义使用 uint32 存储，所以超过 4G 部分的断点位置值会溢出，存储的是一个错误的 checkpoint，重启任务或者 dm-worker 后，使用该值重新解析 binlog/relay log 就会出错，遇到这种情况需要手动进行恢复。恢复方法：
```

File: dev/faq/data-migration.md [hunk 2]
```diff
@@ -19,3 +19,10 @@ category: FAQ
...
+
+- 首先判断出错发生在 relay log 写入还是 binlog replication unit 同步，如果错误发生在 relay log 写入，binlog replication 保存的断点都是正确的情况，可以先停止任务，停止 DM-worker，手动调节 relay meta 的 binlog-position 到 4，重启 DM-worker 重新拉取 relay log ，relay log 写入正常后启动任务会自动从断点继续同步。
```
Comment (GregoryIan) [inline dev/faq/data-migration.md]:
```suggestion
- 首先判断出错发生在 relay log 写入还是 binlog replication/syncer unit 同步，如果错误发生在 relay log 模块，binlog replication/syncer unit 保存的断点都是正确的情况，可以先停止任务，停止 DM-worker，手动调节 relay meta 的 binlog-position 到 4，重启 DM-worker 重新拉取 relay log ，relay log 写入正常后启动任务会自动从断点继续同步。
```

File: dev/faq/data-migration.md [hunk 3]
```diff
@@ -19,3 +19,10 @@ category: FAQ
...
+- 如果 relay log 写入正常，已经写到了下一个文件，错误发生在 binlog replication unit 读取当前超过 4G relay log 文件的一个不合法 position，这时候可以停止任务，手动调节该任务的 checkpoint 信息到该 relay log 的一个合法 position，比如 4，注意需要同时调整 global checkpoint 和每张表的 checkpoint（如果某张表的 checkpoint 是比当前 relay log 更旧的 relay log，则不需要修改）。设置任务的 sage-mode 为 true ，保证可重入执行，这之后就可以重新启动同步任务，观察同步状态。
```
Comment (GregoryIan) [inline dev/faq/data-migration.md]:
```suggestion
- 如果 relay log 写入正常，已经写到了下一个文件，错误发生在 binlog replication/syncer unit 读取当前超过 4G relay log 文件的一个不合法的 binlog position，这时候可以停止任务，手动调节该任务的 binlog position 信息到该 relay log 的一个合法的 binlog position，比如 4，注意需要同时调整 global checkpoint 和每个 table checkpoint 的 binlog position。设置任务的 safe-mode 为 true ，保证可重入执行，这之后就可以重新启动同步任务，观察同步状态。
```

**Comment by GregoryIan** [comment]
LGTM

File: dev/faq/data-migration.md [hunk 4]
```diff
@@ -19,3 +19,10 @@ category: FAQ
 发生 `driver: bad connection` 错误时，通常表示 DM 到下游 TiDB 的数据库连接出现了异常（如网络故障、TiDB 重启等）且当前请求的数据暂时未能发送到 TiDB。
 
 当前版本 DM 发生该类型错误时，需要先使用 `stop-task` 停止任务后再使用 `start-task` 重启任务。后续 DM 会完善对此错误类型的自动重试机制。
+
+## 同步任务中断并包含 `get binlog error ERROR 1236 (HY000)`, `binlog checksum mismatch, data may be corrupted` 等 binlog 获取或解析失败错误
+
+在 DM 进行增量同步过程中，如果遇到了上游超过 4G 的 binlog 文件，并且 DM 在处理这个 binlog 过程中出现了同步中断（包括正常的 pause/stop 任务，异常的原因发生的中断时）, 就可能出现这个错误。原因是 DM 需要保存同步的 binlog position 信息，但是 MySQL binlog position 官方定义使用 uint32 存储，所以超过 4G 部分的 binlog position 的 offset 值会溢出，就会存储的是一个错误的 binlog position，在重启 task 或者 dm-worker 后，需要使用该 binlog position 重新解析 binlog/relay log，进而出现上面的错误。遇到这种情况需要手动进行恢复。恢复方法：
```
Comment (WangXiangUSTC) [inline dev/faq/data-migration.md]:
```suggestion
在 DM 进行增量同步过程中，如果遇到了上游超过 4G 的 binlog 文件，并且 DM 在处理这个 binlog 过程中出现了同步中断（包括正常的 pause/stop 任务，异常的原因导致的中断时）, 就可能出现这个错误。原因是 DM 需要保存同步的 binlog position 信息，但是 MySQL binlog position 官方定义使用 uint32 存储，所以超过 4G 部分的 binlog position 的 offset 值会溢出，就会存储的是一个错误的 binlog position，在重启 task 或者 dm-worker 后，需要使用该 binlog position 重新解析 binlog/relay log，进而出现上面的错误。遇到这种情况需要手动进行恢复，恢复方法：
```

File: dev/faq/data-migration.md [hunk 5]
```diff
@@ -19,3 +19,10 @@ category: FAQ
...
+
+- 首先判断出错发生在 relay log 写入还是 binlog replication/syncer unit 同步，如果错误发生在 relay log 模块，binlog replication/syncer unit 保存的断点都是正确的情况，可以先停止任务，停止 DM-worker，手动调节 relay meta 的 binlog-position 到 4，重启 DM-worker 重新拉取 relay log ，relay log 写入正常后启动任务会自动从断点继续同步。
+- 如果 relay log 写入正常，已经写到了下一个文件，错误发生在 binlog replication/syncer unit 读取当前超过 4G relay log 文件的一个不合法的 binlog position，这时候可以停止任务，手动调节该任务的 binlog position 信息到该 relay log 的一个合法的 binlog position，比如 4，注意需要同时调整 global checkpoint 和每个 table checkpoint 的 binlog position。设置任务的 safe-mode 为 true ，保证可重入执行，这之后就可以重新启动同步任务，观察同步状态，只要同步过这个大于 4G 的文件即可。
```
Comment (WangXiangUSTC) [inline dev/faq/data-migration.md]:
```suggestion
- 如果 relay log 写入正常，已经写到了下一个文件，错误发生在 binlog replication/syncer unit 读取当前超过 4G relay log 文件的一个不合法的 binlog position，这时候可以停止任务，手动调节该任务的 binlog position 信息到该 relay log 的一个合法的 binlog position，比如 4，注意需要同时调整 global checkpoint 和每个 table checkpoint 的 binlog position。设置任务的 safe-mode 为 true ，保证可重入执行，在这之后就可以重新启动同步任务，观察同步状态，只要同步过这个大于 4G 的文件即可。
```

Comment (WangXiangUSTC) [reply dev/faq/data-migration.md]: 需不需要说明下 checkpoint 是存到下游数据库的？还是说默认用户知道这个东西？

**Comment by WangXiangUSTC** [comment]
LGTM

**Pull request merged**